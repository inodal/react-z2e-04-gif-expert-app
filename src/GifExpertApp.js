import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import ReactDOM from 'react-dom';

const GifExpertApp = () => {
    const initialState = ['MotoGP'];
    const [categories, setCategories] = useState(initialState);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />
            
            <ol>
                {
                    categories.map(c => (
                        <GifGrid 
                            key = { c }
                            category = { c } 
                        />
                    ))
                }
            </ol>
        </>
    );

};

export default GifExpertApp;