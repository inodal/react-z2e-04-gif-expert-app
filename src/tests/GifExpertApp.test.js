import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'; // without {} becauseof 'export default'


describe('GifExpertApp component test suite', () => {
    
    test('should show GifExpertApp properly', () => {
    
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show GifExpertApp with custo categories properly', () => {

        const categories = ['Mandalorian', 'Death Star']
    
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
