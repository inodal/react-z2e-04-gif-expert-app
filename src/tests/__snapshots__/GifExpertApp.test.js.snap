// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp component test suite should show GifExpertApp properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp component test suite should show GifExpertApp with custo categories properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Mandalorian"
      key="Mandalorian"
    />
    <GifGrid
      category="Death Star"
      key="Death Star"
    />
  </ol>
</Fragment>
`;
